features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/root/PreferencesRootView.kt

```diff
@@ -162,7 +163,13 @@ fun PreferencesRootView(
             headlineContent = { Text(stringResource(id = CommonStrings.action_signout)) },
             leadingContent = ListItemContent.Icon(IconSource.Resource(CommonDrawables.ic_sign_out)),
             style = ListItemStyle.Destructive,
-            onClick = onSignOutClicked,
+            onClick = {
+                if (state.directLogoutState.canDoDirectSignOut) {
+                    state.directLogoutState.eventSink(DirectLogoutEvents.Logout(ignoreSdkError = false))
```

Comment:
Maybe we could change the `onSignOutClicked` lambda so it takes a `direct: Boolean` param and handle the logic there?

Reply:
Moved to the Node, which also knows the state, so no need for a parameter.